services/src/main/java/org/keycloak/keys/AbstractImportedRsaKeyProviderFactory.java

```diff
@@ -81,9 +82,15 @@ public void validateConfiguration(KeycloakSession session, RealmModel realm, Com
             if (!certificate.getPublicKey().equals(keyPair.getPublic())) {
                 throw new ComponentValidationException("Certificate does not match private key");
             }
+
+            try {
+                certificate.checkValidity();
+            } catch (CertificateException e) {
+                throw new ComponentValidationException("Certificate is not valid");
```

Comment:
```suggestion
                throw new ComponentValidationException("Certificate is not valid", e);
```

Minor: Is it possible to keep the original exception and make sure that when administrator tries to upload the key with expired certificate, the underlying cause is logged as well - at least to server.log, but maybe even to the error message itself if possible, so it is returned in the response? At least, that is what I usually preferred when using things as the user to have more details if possible if something doesn't work :-)

Reply:
:+1: Yes, it's much better. Done now!